### PR TITLE
disable aws params for dolt_sync command run via sql and minor credentialing change

### PR DESCRIPTION
**Pull request by bheni**

### CODE REVIEW

File: go/libraries/doltcore/dbfactory/aws.go [hunk 1]
```diff
@@ -142,6 +142,11 @@ func (fact AWSFactory) newChunkStore(ctx context.Context, nbf *types.NomsBinForm
 	}
 
 	sess := session.Must(session.NewSessionWithOptions(opts))
+	_, err = sess.Config.Credentials.Get()
```
Comment (bheni) [inline go/libraries/doltcore/dbfactory/aws.go:145]: Doing this check here and returning the error improves error messaging.

File: go/libraries/doltcore/dbfactory/aws.go [hunk 2]
```diff
@@ -188,6 +193,11 @@ func awsConfigFromParams(params map[string]interface{}) (session.Options, error)
 		opts.Profile = val.(string)
 	}
 
+	filePath, ok := params[AWSCredsFileParam]
+	if ok && len(filePath.(string)) != 0 && awsCredsSource == RoleCS {
+		awsCredsSource = FileCS
+	}
+
```
Comment (bheni) [inline go/libraries/doltcore/dbfactory/aws.go:200]: If a credential file is provided then change the source to FileCS unless it has been explicitly changed from the default of RoleCS

File: go/libraries/doltcore/sqle/dfunctions/dolt_backup.go [hunk 3]
```diff
@@ -122,6 +129,18 @@ func DoDoltBackup(ctx *sql.Context, args []string) (int, error) {
 			return statusErr, err
 		}
 
+		credsFile, _ := sess.GetSessionVariable(ctx, dsess.AwsCredsFileKey)
```
Comment (zachmu) [inline go/libraries/doltcore/sqle/dfunctions/dolt_backup.go:132]: Err handling for when these vars are absent?

**Review by zachmu — APPROVED** [review]
LGTM

**Pull request merged**